lib/core-classes/DeviceService.ts

```diff
@@ -86,9 +86,20 @@ export class DeviceService {
       }
 
       const deviceDocuments = this.formatDevicesContent(devices, document);
+      const enrichedDocuments = [];
+
+      for (const deviceDocument of deviceDocuments) {
+        const response = await global.app.trigger(
+          'device-manager:device:attach-tenant:before',
+          { index: document.tenantId, device: deviceDocument }
+        );
+        
+        const deviceData = response && response.device ? response.device : deviceDocument;
```

Comment:
Same remark as on the other PR, it's the developer responsibility to return a correct payload in the pipe